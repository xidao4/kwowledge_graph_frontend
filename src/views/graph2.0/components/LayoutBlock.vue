<template>
    <div class="block">
        <div v-if="currentGraphId === graphIds.force" style="width: 100%;">
<!--            <div class="inline-title">-->
<!--                <span class="label">当前布局</span>-->
<!--                <span class="title">力引导布局</span>-->
<!--            </div>-->
            <a-row type="flex" justify="space-between" align="middle">
                <a-col :xs="24" :sm="24" :md="7" :lg="7" :xl="7">
                    <span class="label">当前布局</span>
                </a-col>
                <a-col :xs="24" :sm="24" :md="16" :lg="16" :xl="16">
                    <span class="title">力引导布局</span>
                </a-col>
            </a-row>
            <a-row type="flex" justify="space-between" align="middle">
                <a-col :xs="24" :sm="24" :md="7" :lg="7" :xl="7">
                    <span class="label">作用力</span>
                </a-col>
                <a-col :xs="24" :sm="24" :md="6" :lg="8" :xl="8">
                    <a-slider v-model="nodeStrength" :min="-50" :max="50" class="slider"/>
                </a-col>
                <a-col :xs="0" :sm="0" :md="10" :lg="8" :xl="8">
                    <a-input-number v-model="nodeStrength" :min="-50" :max="50"/>
                </a-col>
            </a-row>
<!--            <div class="inline-title" style="margin-top: 5px">-->
<!--                <a-row>-->
<!--                    <a-col span="7">-->
<!--                        <span class="label">作用力</span>-->
<!--                    </a-col>-->
<!--                    <a-col span="12">-->
<!--                        <a-slider v-model="nodeStrength" :min="-50" :max="50" class="slider"/>-->
<!--                    </a-col>-->
<!--                    <a-col span="5" offset="1">-->
<!--                        <a-input-number v-model="nodeStrength" :min="-50" :max="50"/>-->
<!--                    </a-col>-->
<!--                </a-row>-->
<!--            </div>-->
        </div>
        <div class="inline-title" v-if="currentGraphId === graphIds.typesetting">
            <a-select :default-value="currentSetLayout" style="width: 200px" @change="handleLayoutChange">
                <a-select-opt-group label="力引导布局">
                    <a-select-option :value="item.key" v-for="(item, index) in forceLayout" :key="index">
                        {{item.value}}
                    </a-select-option>
                </a-select-opt-group>
                <a-select-opt-group label="一般布局">
                    <a-select-option :value="item.key" v-for="(item, index) in layoutType" :key="index">
                        {{item.value}}
                    </a-select-option>
                </a-select-opt-group>
            </a-select>
        </div>
    </div>
</template>

<script>
    import { mapGetters } from 'vuex';
    import ARow from "ant-design-vue/es/grid/Row";
    import ACol from "ant-design-vue/es/grid/Col";
    export default {
        name: "LayoutBlock",
        components: {ACol, ARow},
        data(){
            return {
                nodeStrength: 30
            }
        },
        computed: {
            ...mapGetters([
                'currentGraphId',
                'graphIds',
                'currentSetLayout',
                'forceLayout',
                'layoutType'
            ]),
        },
        methods: {
            handleLayoutChange(value) {
                console.log(`selected ${value}`);
            },
        }
    }
</script>

<style scoped>
.block {
    min-height: 100px;
    padding: 2vw;
    display: flex;
    justify-content: space-evenly;
    align-items: center;
}
.inline-title {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    align-items: center;
}
.title {
    font-size: 18px;
    font-weight: bold;
    color: #D99CA8;
    width: 75%;
}
.label{
    /*margin-right: 5px;*/
    width: 25%;
    font-size: 13px;
}
.inline-title .slider{
    width: 75%;
}
@media screen and (max-width:800px){
    .block {
        justify-content: flex-start;
    }
    .inline-title {
        display: flex;
        flex-direction: column;
        justify-content: flex-start;
        align-items: flex-start;
    }
    .inline-title .label{
        margin-right: 0;
    }
    .inline-title .slider{
        /*width: 100%;*/
    }
}
</style>
