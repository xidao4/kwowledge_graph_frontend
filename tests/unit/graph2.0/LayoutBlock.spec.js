import { mount, createLocalVue } from '@vue/test-utils'
import LayoutBlock from '@/views/graph2.0/components/LayoutBlock.vue'
import Antd from 'ant-design-vue';
import Vuex from 'vuex';

const localVue = createLocalVue();
localVue.use(Antd);
localVue.use(Vuex);


describe('LayoutBlock', () => {
    let mutations;
    let getters;
    let store;
    let wrapper;

    beforeEach(() => {
        mutations = {
            set_currentSetLayout: jest.fn(),
            set_currentCombos: jest.fn(),
            set_currentShowCombos: jest.fn(),
        };
        getters = {
            currentGraphId: () => 'typesetting',
            graphIds: () => {
                return {
                    force: 'force',
                    typesetting: 'typesetting'
                }
            },
            currentSetLayout: () => 'radial',
            forceLayout: () => {
                return [
                    {
                        key: 'gForce',
                            value: '经典力导向布局(gForce)'
                    },
                    {
                        key: 'force',
                            value: '经典力导向布局(force)'
                    },
                    {
                        key: 'forceAtlas2',
                            value: 'FA2力导向布局(forceAtlas2)'
                    },
                    {
                        key: 'fruchterman',
                            value: 'Fruchterman布局(fruchterman)'
                    },
                ]
            },
            layoutType: () => {
                return [
                    {
                        key: 'random',
                        value: '随机布局(random)'
                    },
                    {
                        key: 'circular',
                        value: '环形布局(circular)'
                    },
                    {
                        key: 'radial',
                        value: '辐射状布局(Radial)'
                    },
                    {
                        key: 'mds',
                        value: '高维数据降维算法布局(mds)'
                    },
                    {
                        key: 'dagre',
                        value: '层次布局(dagre)'
                    },
                    {
                        key: 'concentric',
                        value: '同心圆布局(concentric)'
                    },
                    {
                        key: 'grid',
                        value: '格子布局(grid)'
                    },
                ]
            },
            currentGraph: () => {
                return {
                    data: jest.fn(),
                    updateLayout: jest.fn()
                }
            },
            forceGraph: () => {
                return {
                    data: jest.fn(),
                    updateLayout: jest.fn(() => {
                        console.log('force update')
                    }),
                    render: jest.fn()
                }
            },
            typesettingGraph: () => {
                return {
                    data: jest.fn(),
                    updateLayout: jest.fn(() => {
                        console.log('typesetting update')
                    }),
                    render: jest.fn()
                }
            },
            currentCombos: () => ["c0", "c1"],
            currentShowCombos: () => false,
            currentShowGraphData: () => {
                return {
                    typesetting: {},
                    force: {}
                }
            }
        };
        store = new Vuex.Store({
            state: {},
            mutations,
            getters,
        });
        wrapper = mount(LayoutBlock, {
            store,
            localVue
        });
    });

    it('layoutBlock snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    });

    // it('handleLayoutChange',  () => {
    //     let outputData = "";
    //     let storeLog = inputs => (outputData += inputs);
    //     console["log"] = jest.fn(storeLog);
    //     let select = wrapper.find('.a-select');
    //     select.element.value = 'radial';
    //     select.trigger('change');
    //     expect(outputData).toBe("force updatetypesetting update");
    //     // expect(getters.currentGraph().updateLayout).toHaveBeenCalled();
    //     // expect(getters.forceGraph().updateLayout()).toBe('update');
    // });

    // it('input修改作用力', () => {
    //     let input = wrapper.find('input');
    //     input.element.value = '1';
    //     input.trigger('input')
    // })

    it('changeGroup', () => {
        let button = wrapper.find('.ant-switch');
        button.element.value = '1';
        button.trigger('click')
    });

    it('handleLayoutChange', () => {
        // TODO 监听到ant组件
        let select = wrapper.find('.ant-select-selection__rendered');
        select.trigger('change')
    })
});

