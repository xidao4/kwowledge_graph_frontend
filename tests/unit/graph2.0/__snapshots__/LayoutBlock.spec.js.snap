// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutBlock layoutBlock snapshot 1`] = `
<div class="block">
  <div style="width: 100%; display: none;">
    <div class="ant-row-flex ant-row-flex-space-between ant-row-flex-middle">
      <div class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-7 ant-col-lg-7 ant-col-xl-7"><span class="label">当前布局</span></div>
      <div class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-16 ant-col-lg-16 ant-col-xl-16"><span class="title">力引导布局</span></div>
    </div>
    <div class="ant-row-flex ant-row-flex-space-between ant-row-flex-middle">
      <div class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-7 ant-col-lg-7 ant-col-xl-7"><span class="label">作用力</span></div>
      <div class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-6 ant-col-lg-8 ant-col-xl-8">
        <div tabindex="-1" class="slider ant-slider">
          <div class="ant-slider-rail"></div>
          <div class="ant-slider-track" style="left: 0%; width: 55.00000000000001%;"></div>
          <div class="ant-slider-step"></div>
          <div role="slider" tabindex="0" aria-valuemin="-10" aria-valuemax="10" aria-valuenow="1" class="ant-slider-handle" style="left: 55.00000000000001%; transform: translateX(-50%);"></div>
          <div class="ant-slider-mark"></div>
        </div>
      </div>
      <div class="ant-col ant-col-xs-0 ant-col-sm-0 ant-col-md-10 ant-col-lg-8 ant-col-xl-8">
        <div class="ant-input-number">
          <div class="ant-input-number-handler-wrap"><span class="ant-input-number-handler ant-input-number-handler-up " unselectable="unselectable" role="button" aria-label="Increase Value"><i aria-label="icon: up" class="anticon anticon-up ant-input-number-handler-up-inner"><svg viewBox="64 64 896 896" data-icon="up" width="1em" height="1em" fill="currentColor" aria-hidden="true" focusable="false" class=""><path d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 0 0 140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"></path></svg></i></span><span class="ant-input-number-handler ant-input-number-handler-down " unselectable="unselectable" role="button" aria-label="Decrease Value"><i aria-label="icon: down" class="anticon anticon-down ant-input-number-handler-down-inner"><svg viewBox="64 64 896 896" data-icon="down" width="1em" height="1em" fill="currentColor" aria-hidden="true" focusable="false" class=""><path d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"></path></svg></i></span></div>
          <div class="ant-input-number-input-wrap"><input role="spinbutton" aria-valuemin="-10" aria-valuemax="10" aria-valuenow="1" autocomplete="off" max="10" min="-10" step="1" class="ant-input-number-input"></div>
        </div>
      </div>
    </div>
  </div>
  <div style="width: 100%;">
    <div><span class="margin-right-s label">
                布局
            </span>
      <div tabindex="0" class="ant-select ant-select-enabled" style="width: 85%;">
        <div role="combobox" aria-autocomplete="list" aria-haspopup="true" aria-controls="test-uuid" class="ant-select-selection ant-select-selection--single">
          <div class="ant-select-selection__rendered">
            <div unselectable="on" class="ant-select-selection__placeholder" style="display: block; user-select: none;">自定义布局</div>
          </div><span unselectable="on" class="ant-select-arrow" style="user-select: none;"><i aria-label="icon: down" class="anticon anticon-down ant-select-arrow-icon"><svg viewBox="64 64 896 896" data-icon="down" width="1em" height="1em" fill="currentColor" aria-hidden="true" focusable="false" class=""><path d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"></path></svg></i></span>
        </div>
      </div>
    </div>
    <div class="margin-top"><span class="margin-right-s label">
                分组
            </span> <button type="button" role="switch" class="ant-switch"><span class="ant-switch-inner"><i aria-label="icon: close" class="anticon anticon-close"><svg viewBox="64 64 896 896" data-icon="close" width="1em" height="1em" fill="currentColor" aria-hidden="true" focusable="false" class=""><path d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"></path></svg></i></span></button></div>
  </div>
</div>
`;
